integrations/pooltogether-v4/components/form-yield-source-prize-pool-deposit.tsx

```diff
@@ -0,0 +1,145 @@
+import { useUsdcApproval } from '@/actions/pooltogether-v4/hooks/use-usdc-approval'
+import { useUserBalanceDeposit } from '@/actions/pooltogether-v4/hooks/use-user-balance-deposit'
+import { usePoolTogetherPrizePoolDepositTo } from '@/actions/pooltogether-v4/pooltogether-v4-wagmi'
+import { PRIZE_POOL_CONTRACT } from '@/actions/pooltogether-v4/prize-pool-contract-list'
+import { USDC_CONTRACT } from '@/actions/pooltogether-v4/usdc-contract-list'
+import { Checkbox } from '@/components/ui/checkbox'
+
+export function FormDeposit() {
+  const { address } = useAccount()
+  const prizePoolAddress = useLoadContractFromChainId(PRIZE_POOL_CONTRACT)
+  const usdcAddress = useLoadContractFromChainId(USDC_CONTRACT)
+
+  const { data: decimals } = useErc20Decimals({ address: usdcAddress })
+  const POWER: any = decimals != undefined ? BigNumber.from(10).pow(decimals) : BigNumber.from(10).pow(6)
+
+  const userBalance = useUserBalanceDeposit()
+  const isApproved = useUsdcApproval(userBalance)
+
+  const [depositAmount, setDepositAmount] = useState('')
+  const debouncedDepositAmount = useDebounce((POWER as any) * Number(depositAmount), 500)
+
+  const [isChecked, setIsChecked] = useState(false)
+  const [approvalAmount, setApprovalAmount] = useState<BigNumber>(BigNumber.from(0))
+  const [submitDeposit, setSubmitDeposit] = useState<boolean>(false)
+  const [isValidAmount, setValidAmount] = useState<boolean>(true)
+
+  // @ts-ignore
+  const { data, write: depositToken } = usePoolTogetherPrizePoolDepositTo({
+    address: prizePoolAddress,
+    args: [address, debouncedDepositAmount],
+    enabled: isApproved && Boolean(debouncedDepositAmount),
+    overrides: {
+      gasLimit: BigNumber.from(750000),
+    },
+  })
+
+  const { isLoading } = useWaitForTransaction({
+    hash: data?.hash,
+  })
+
+  // @ts-ignore
+  const { data: approveData, write: approval } = useErc20Approve({
+    address: usdcAddress,
+    args: [prizePoolAddress, approvalAmount],
+    overrides: {
+      gasLimit: BigNumber.from(750000),
+    },
+  })
+
+  const { isLoading: loadApprove, isSuccess: successApprove } = useWaitForTransaction({
+    hash: approveData?.hash,
+  })
+
+  useEffect(() => {
+    isChecked ? setApprovalAmount(ethers.constants.MaxInt256) : setApprovalAmount(BigNumber.from(debouncedDepositAmount))
+  }, [isChecked])
+
+  useEffect(() => {
+    if (successApprove || submitDeposit) {
+      depositToken?.()
+      setSubmitDeposit(false)
+    }
+  }, [successApprove, submitDeposit])
+
+  const handleSubmit = (event: any) => {
+    event.preventDefault()
+    if (parseFloat(depositAmount) >= 2.0) {
+      if (!isApproved) {
+        approval?.()
+      } else {
+        setSubmitDeposit(true)
+      }
+    } else {
+      setValidAmount(false)
+    }
+  }
+  const handleChange = (event: any) => {
+    const value = event.target.value
+    const regex = /^[0-9]*\.?[0-9]*$/
```

Comment:
I don't get exactly what you're trying to do here

Reply:
got it, instead of doing it you can just change the input type from `text` to `number` and set the min value to 0 and you'll get the same result